src/services/document-service.js

```diff
@@ -447,6 +447,32 @@ class DocumentService {
     }
   }
 
+  async validateDocument(documentId) {
+    const doc = await this.documentStore.getDocumentById(documentId);
+    for (const section of doc.sections) {
+      try {
+        validateSection(section, this.pluginRegistry);
+      } catch (error) {
+        error.documentId = doc._id;
+        error.sectionKey = section.key;
+        throw error;
```

Comment:
What would you say about collecting all validation errors on the document and revision and throwing an error at the end with the full report?
Maybe something like `validationError.cases.push({ documentId/documentRevisionId, sectionKey, error })`?
Otherwise the validation process has to be repeatedly run until all errors are fixed.